test/e2e/apps/statefulset.go

```diff
@@ -843,6 +847,75 @@ var _ = SIGDescribe("StatefulSet", func() {
 				return nil
 			}, framework.StatefulPodTimeout, 2*time.Second).Should(BeNil())
 		})
+
+		It("should have a working scale subresource", func() {
```

Comment:
> I am able to test it manually using kubectl proxy and curl

all `kubectl proxy` does is add credentials from your local kubeconfig file to requests made to `http://localhost:8001/...` and pass the request along to the API server. if this is working for you, it must mean you have `apps/v1beta2` enabled on your API server, and could make the same request directly against the API server, without `kubectl proxy` being involved at all.

if all you want to do in the e2e is to submit a request to an API endpoint, there's no need to use curl, or to run `kubectl proxy`.

I'd expect the e2e test to do something more like this:
```
			By("getting scale subresource")
			scale := &extensionsv1beta1.Scale{}
			err := c.AppsV1beta2().RESTClient().Get().AbsPath("/apis/apps/v1beta1").Namespace(ns).Resource("statefulsets").Name(ssName).SubResource("scale").Do().Into(scale)
			if err != nil {
				framework.Failf("Failed to get scale subresource: %v", err)
			}
			Expect(scale.Spec.Replicas).To(Equal(1))
			Expect(scale.Status.Replicas).To(Equal(1))

			By("updating a scale subresource")
			// unconditionally update to 2 replicas
			scale.ResourceVersion = ""
			scale.Spec.Replicas = 2
			scaleResult := &extensionsv1beta1.Scale{}
			err := c.AppsV1beta2().RESTClient().Put().AbsPath("/apis/apps/v1beta1").Namespace(ns).Resource("statefulsets").Name(ssName).SubResource("scale").Body(scale).Do().Into(scaleResult)
			if err != nil {
				framework.Failf("Failed to put scale subresource: %v", err)
			}
			Expect(scaleResult.Spec.Replicas).To(Equal(2))

			By("verifying the stateful set spec was modified")
			ss, err := c.AppsV1beta2().StatefulSets(ns).Get(ssName, metav1.GetOptions{})
			if err != nil {
				framework.Failf("Failed to get stateful set resource: %v", err)
			}
			Expect(ss.Spec.Replicas).To(Equal(2))
```